example/App.vue

```diff
@@ -410,29 +410,15 @@
 </template>
 
 <script lang="ts">
-import Vue from "vue";
+import { defineComponent, computed, toRefs, ToRefs, Ref, provide, reactive } from "vue";
 import * as Phosphor from "@/entry";
-import { ExtendedVue } from "vue/types/vue";
-import { IconProps } from "@/lib/types";
-import { IconComputed } from "@/lib/types";
 
-type VueIcon = ExtendedVue<Vue, {}, {}, IconComputed, IconProps> & {
-  options: { name: string };
-};
-
-function isIcon(candidate: any): candidate is VueIcon {
-  return candidate.options && candidate.options.name;
-}
-
-const Icon = Object.values(Phosphor).reduce((components, Icon) => {
-  if (isIcon(Icon)) return { ...components, [Icon.options!!.name]: Icon };
-  return components;
-}, {});
-
-const icons = Object.values(Icon);
+// Technically more fragile than the former type guard, but Vue 3 doesn't 
+const { default: _, ...Icon } = Phosphor;
+const allIcons = Object.keys(Icon);
```

Comment:
Pardon the truncated comment here. The gist is that some of the necessary types for the type guard are private in Vue 3, and rather than putting two more hours into sorting the mess out I just opted for simply omitting `default` from the plugin import and _assuming_ the remaining members are all icon components.

If this were something a user would be exposed to I'd solve the typing problem, but since this is only a demo site I didn't think it too much of a burden to keep in sync with any new exports from `entry.ts`.

Reply:
Stellar to see a PR like this, thanks for putting in the time! On first glance it seems solid to me, but I have exactly zero experience with Vue3 and little enough with Vue2 to start. That being said, I'll give it a while on my end and see what I see.

I'm not too bothered about relaxing the type checks in the example app, but I'd like to keep explicit types in components if it's doable in v3. Personally, types feel a bit clunky in Vue so I feel your decision to forge ahead for now. And yes, `entry.ts` only registers the members of `lib/index.ts`, which are all icon components. Also open to suggestions when it comes to making this more idiomatic Vue. I am very much a React person writing what they think is a passable Vue port 😄 .

Reply:
Stellar to see a PR like this, thanks for putting in the time! On first glance it seems solid to me, but I have exactly zero experience with Vue3 and little enough with Vue2 to start. That being said, I'll give it a while on my end and see what I see.

I'm not too bothered about relaxing the type checks in the example app, but I'd like to keep explicit types in components if it's doable in v3. Personally, types feel a bit clunky in Vue so I feel your decision to forge ahead for now. And yes, `entry.ts` only registers the members of `lib/index.ts`, which are all icon components. Also open to suggestions when it comes to making this more idiomatic Vue. I am very much a React person writing what they think is a passable Vue port 😄 .

Reply:
Yes, `entry.ts` only registers the exported members of `lib/index.ts` which are all icon components, so your guard is good for now.